autogluon/task/tabular_prediction/predictor.py

```diff
@@ -412,6 +424,50 @@ def transform_features(self, dataset=None, model=None, base_models=None):
             dataset_transformed.index = original_indices
         return dataset_transformed
 
+    # TODO: Add support for DataFrame input and support for DataFrame output
+    #  Add support for retaining DataFrame/Series indices in output
+    #  Add as_pandas parameter
+    def transform_labels(self, labels, inverse=False, proba=False):
+        """
+        Transforms dataset labels to the internal label representation.
+        Regression problems do not differ between original and internal representation, and thus this method will return the provided labels.
+        Warning: When `inverse=False`, it is possible for the output to contain NaN label values in multiclass problems if the provided label was dropped during training.
+
+        Parameters
+        ----------
+        labels : `numpy.ndarray` or `pandas.Series`
+            Labels to transform.
+            If `proba=False`, an example input would be the output of `predictor.predict(test_data)`.
+            If `proba=True`, an example input would be the output of `predictor.predict_proba(test_data)`.
+        inverse : boolean, default = False
+            When `True`, the input labels are treated as being in the internal representation and the original representation is outputted.
+        proba : boolean, default = False
+            When `True`, the input labels are treated as probabilities and the output will be the internal representation of probabilities.
+                In this case, it is expected that `labels` be a `numpy.ndarray`.
+                If the `problem_type` is multiclass:
+                    The input column order must be equal to `predictor.class_labels`.
+                    The output column order will be equal to `predictor.class_labels_internal`.
+                    if `inverse=True`, the same logic applies, but with input and output columns interchanged.
+            When `False`, the input labels are treated as actual labels and the output will be the internal representation of the labels.
+                In this case, it is expected that `labels` be a `pandas.Series` or `numpy.ndarray`.
```

Comment:
Can you clarify what is a use-case where `proba=True`?

Reply:
Imagine someone uses transform_features and transform_labels to generate the internal representation, then trains a model on that data. They then pass in their test data through transform_features to get prediction probabilities. The pred_proba is currently in the internal representation and they want the probabilities in the original label representation, so they pass it in as predictor.transform_labels(internal_pred_proba, inverse=True, proba=True) to get the original label pred_proba.

proba=True, inverse=False would be quite niche, but for example if the user had their own custom model predictions and they wanted to ensemble them with an internal model to create yet another model, they could convert their external proba estimates to the internal one to align.